core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -188,7 +204,14 @@ private ScriptEngineService getScriptEngineServiceForFileExt(String fileExtensio
         return scriptEngineService;
     }
 
-
+    void setMaxCompilationsPerMinute(Integer newMaxPerMinute) {
+        if (newMaxPerMinute != null) {
```

Comment:
It'd be nice to have a note about why you have to handle `null` here and what null means. It looks like `null` means "this cluster state change didn't change the setting".